auth_keycloak/wizard/keycloak_sync_wiz.py

```diff
@@ -0,0 +1,300 @@
+    """Export Odoo users to Keycloak.
+
+    Usually Keycloak is already populated w/ your users base.
+    Many times this will come via LDAP, AD, pick yours.
+
+    Still, you might need to push some users to Keycloak on demand,
+    maybe just for testing.
+
+    If you need this, this is the wizard for you ;)
+    """
+
+    _name = 'auth.keycloak.create.wiz'
+    _inherit = 'auth.keycloak.sync.mixin'
+
+    user_ids = fields.Many2many(
+        comodel_name='res.users',
+        default=lambda self: self.env.context.get('active_ids'),
+    )
+
+    def _validate_setup(self):
+        super(KeycloakCreateWiz, self)._validate_setup()
+        if not self.user_ids:
+            raise exceptions.UserError(
+                _('No user selected')
+            )
+
+    def _validate_response(self, resp, no_json=False):
+        # When Keycloak detects a clash on non-unique values, like emails,
+        # it raises:
+        # `HTTPError: 409 Client Error: Conflict for url: `
+        # http://keycloak:8080/auth/admin/realms/master/users
+        if resp.status_code == 409:
+            detail = ''
+            if resp.content and resp.json().get('errorMessage'):
+                # ie: {"errorMessage":"User exists with same username"}
+                detail = '\n' + resp.json().get('errorMessage')
+            raise exceptions.UserError(_(
+                'Conflict on user values. '
+                'Please verify that all values supposed to be unique '
+                'are really unique. %(detail)s'
+            ) % {'detail': detail})
+        super(KeycloakCreateWiz, self)._validate_response(
+            resp, no_json=no_json)
+
+    def _get_or_create_user(self, token, odoo_user):
+        """Lookup for given user on Keycloak: create it if missing.
+
+        :param token: valid auth token from Keycloak
+        :param odoo_user: res.users record
+        """
+        odoo_key = self.login_match_key.split(':')[1]
+        keycloak_user = self._get_users(
+            token, search=odoo_user.mapped(odoo_key)[0])
+        if keycloak_user:
+            if len(keycloak_user) > 1:
+                # TODO: warn user?
+                pass
+            return keycloak_user[0]
+        else:
+            values = self._create_user_values(odoo_user)
+            keycloak_user = self._create_user(token, **values)
+        return keycloak_user
+
+    def _create_user_values(self, odoo_user):
+        """Prepare Keycloak values for given Odoo user."""
+        values = {
+            'username': odoo_user.login,
+            'email': odoo_user.email,
+        }
+        if 'firstname' in odoo_user.partner_id:
+            # partner_firstname installed
+            firstname = odoo_user.partner_id.firstname
+            lastname = odoo_user.partner_id.lastname
+        else:
+            # yeah, I know, it's not perfect... you can override it ;)
+            firstname, lastname = odoo_user.name.split(' ')
```

Comment:
We can't guarantee the name includes a space, so we should find a way that it doesn't trigger an error here if name includes :
- no space
- only one space
- multiple space

Reply:
Actually you provided the solution, so you can also implement it cherry-picking 21edcbf8b1cec8943f90790fb8df7510df796016 :sweat_smile:

Reply:
@grindtildeath done (included and squashed, was already coded in there actually ;))